lexer/src/charclass.rs

```diff
@@ -48,3 +48,11 @@ pub fn is_escapable_char(c: char) -> bool {
         _   => true,
     }
 }
+
+pub fn is_ascii_octaldigit(c: char) -> bool {
+    const RADIX:u32 = 8;
+    match c {
+        c if c.to_digit(RADIX) < Some(8) => true,
+        _ => false
+    }
+}
```

Comment:
is this fn ok here ?

Reply:
Yeah, looks good. I think you can pattern match on the `Option` here, too:

```suggestion
    match c.to_digit(8) {
        Some(_) => true,
        None    => false,
    }
}
```

Reply:
nice!

Reply:
Actually now that I think about it we could also do `c.to_digit(8).is_some()`